packages/cli/test/unit/commands/list.test.ts

```diff
@@ -0,0 +1,179 @@
+      useProject({
+        ...defaultProject,
+        id: 'prj_Am19DF8JBL9g89tn4RdDVD59axFi',
+        name: 'prj_Am19DF8JBL9g89tn4RdDVD59axFi',
+      });
+      const deployment = useDeployment({ creator: user });
+
+      client.setArgv('--all');
+      await list(client);
+
+      const output = await readOutputStream(client);
+
+      const { project, org } = getDataFromIntro(output.split('\n')[0]);
+      const header: string[] = formatTable(output.split('\n')[2]);
+      const data: string[] = formatTable(output.split('\n')[3]);
+      data.splice(2, 1);
+
+      expect(project).toBeUndefined();
+      expect(org).toEqual(teamSlug);
+      expect(header).toEqual([
+        'project',
+        'latest deployment',
+        'state',
+        'age',
+        'username',
+      ]);
+      expect(data).toEqual([
+        deployment.url,
+        stateString(deployment.state || ''),
+        user.name,
+      ]);
+    } finally {
+      process.chdir(originalCwd);
+    }
+  });
+  it('should get the deployments for a specified project', async () => {
+    const cwd = fixture('with-team');
+    try {
+      process.chdir(cwd);
+
+      const user = useUser();
+      useTeams('team_MtLD9hKuWAvoDd3KmiHs9zUg');
+      useProject({
+        ...defaultProject,
+        id: 'prj_Am19DF8JBL9g89tn4RdDVD59axFi',
+        name: 'prj_Am19DF8JBL9g89tn4RdDVD59axFi',
+      });
+      const deployment = useDeployment({ creator: user });
+
+      client.setArgv(deployment.name);
+      await list(client);
+
+      const output = await readOutputStream(client);
+
+      const { org } = getDataFromIntro(output.split('\n')[0]);
+      const header: string[] = formatTable(output.split('\n')[2]);
+      const data: string[] = formatTable(output.split('\n')[3]);
+      data.splice(2, 1);
+
+      expect(org).toEqual(teamSlug);
+
+      expect(header).toEqual([
+        'project',
+        'latest deployment',
+        'state',
+        'age',
+        'username',
+      ]);
+      expect(data).toEqual([
+        deployment.url,
+        stateString(deployment.state || ''),
+        user.name,
+      ]);
+    } finally {
+      process.chdir(originalCwd);
+    }
+  });
+});
+
+function getDataFromIntro(output: string): {
+  project: string | undefined;
+  org: string | undefined;
+} {
+  const project = output.match(/(?<=Deployments for )(.*)(?= under)/);
+  const org = output.match(/(?<=under )(.*)(?= \[)/);
+
+  return {
+    project: project?.[0],
+    org: org?.[0],
+  };
+}
+
+function formatTable(output: string): string[] {
+  return output
+    .trim()
+    .replace(/ {3} +/g, ',')
+    .split(',');
+}
+
+function readOutputStream(client: MockClient): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const chunks: Buffer[] = [];
+    client.stderr.resume();
+    client.stderr.on('data', chunk => {
+      chunks.push(chunk);
+      if (chunks.length === 3) {
+        resolve(chunks.toString().replace(/,/g, ''));
```

Comment:
The timeout below will trigger after this resolve happens. I think we should be careful with promise resolution. I don't think it will matter in this specific case, but generally it's a good idea to prevent resolving and rejecting the same promise.

You could keep a reference to the timeout handle when it's created:

```
const timeoutHandle = setTimeout(...);
```

Then clear it here when resolving the promise:

```
clearTimeout(timeoutHandle);
```